compendium/DeclarativeServices/test/gtest/TestFactoryTarget.cpp

```diff
@@ -0,0 +1,582 @@
+
+        // Create the ServiceA~1 configuration object. Specify a two dynamic targets
+        // so that the test::ServiceBInt dependency for  ServiceA~1 will only be
+        // satisfied by ServiceB~123 instance and the test::ServiceCInt will only 
+        // be satisfied by ServiceC~123
+        cppmicroservices::AnyMap props;
+        props["test::ServiceBInt"] = std::string("(ServiceBId=ServiceB~123)");
+        props["test::ServiceCInt"] = std::string("(ServiceCId=ServiceC~123)");
+        auto service = getFactoryService<test::ServiceAInt>("ServiceA", "1", props);
+        ASSERT_TRUE(service);
+
+        // Clean up
+        serviceBReg.Unregister();
+        serviceCReg.Unregister();
+        testBundle.Stop();
+    }
+    /* tFactoryTarget.multipleTargetsExistsAfter.
+     * ServiceA, ServiceB and ServiceC are all factory instances.
+     * ServiceA~1 is dependent on ServiceB~123 and ServiceC~123
+     * ServiceB~123 and ServiceC~123 are registered after the bundle containing ServiceA is
+     * started.
+     */
+
+    TEST_F(tFactoryTarget,multipleTargetsExistAfter)
+    {
+        // Install and start the bundle containing the ServiceA factory.
+        // DS is now listening for factory configuration objects of the form
+        // ServiceA~<instance>
+        test::InstallLib(context, "TestBundleDSFAC2");
+        cppmicroservices::Bundle testBundle = StartTestBundle("TestBundleDSFAC2");
+
+        // Create the ServiceA~1 configuration object. Specify a dynamic target
+        // so that the test::ServiceBInt dependency for  ServiceA~1 will only be
+        // satisfied by ServiceB~123 instance.
+        cppmicroservices::AnyMap props;
+        props["test::ServiceBInt"] = std::string("(ServiceBId=ServiceB~123)");
+        props["test::ServiceCInt"] = std::string("(ServiceCId=ServiceC~123)");
+        auto service = getFactoryService<test::ServiceAInt>("ServiceA", "1", props);
+        // Service cannot be instantiated because it's dependencies are not yet satisfied.
+        ASSERT_TRUE(!service);
+
+        // Create a service tracker for ServiceA.
+        std::unique_ptr<cppmicroservices::ServiceTracker<test::ServiceAInt>> tracker(
+            new cppmicroservices::ServiceTracker<test::ServiceAInt>(context, nullptr));
+        tracker->Open();
+
+        // Register the ServiceB~123 factory service with a mock implementation
+        auto serviceBReg = registerSvc<test::ServiceBInt, MockServiceBImpl>("ServiceB~123", "ServiceBId", "ServiceB~123");
+
+        // Register the ServiceC~123 factory service with a mock implementation
+        auto serviceCReg = registerSvc<test::ServiceCInt, MockServiceCImpl>("ServiceC~123", "ServiceCId", "ServiceC~123");
+  
+        auto serviceA = tracker->WaitForService(std::chrono::milliseconds(500));
+        ASSERT_TRUE(serviceA) << "ServiceA not registered";
+
+        // Clean up
+        tracker->Close();
+        serviceBReg.Unregister();
+        testBundle.Stop();
+    }
```

Comment:
carriage return

Reply:
Not sure what you mean.

Reply:
New line is what I meant. sorry for confusion.

Reply:
Will fix